downloader/src/main/kotlin/DownloadException.kt

```diff
@@ -20,7 +20,7 @@
 package com.here.ort.downloader
 
 class DownloadException : Exception {
-    constructor(cause: Exception): super(cause)
-    constructor(message: String): super(message)
-    constructor(message: String, cause: Exception?): super(message, cause)
+    constructor(cause: Exception?): super(cause)
+    constructor(message: String?): super(message)
```

Comment:
Does this make sense? If I call `constructor(null)` now the compiler cannot know which constructor to call, right?

Reply:
Uhm, good point. I came across this as I was trying to do

        } catch (e: IOException) {
            throw ScanException(e.message)
        }

But the the compiler complained that `e.message` is nullable. Any better idea how to fix that? Should we simply drop one of the single-argument constructors?

Reply:
Well, actually I think I do not want anyone to initialize the exception with `null`, so it is somehow ok to me if it does not work :) I didn't try, but if this works it should be fine, and looking at your code sample it should:

```
fun test(arg: String?) = ScanException(arg)
```

Reply:
Btw, these two constructors should be enough, also for Java Interop, or did I miss something?

```kotlin
    constructor(cause: Exception): super(cause)
    constructor(message: String, cause: Exception? = null): super(message, cause)
```

Reply:
That should be indeed enough. I'll create another PR.

Reply:
Interestingly, it turns out this does not work: Besides the fact that the `cause` should be a `Throwable`, not an `Exception`, Throwables have their `cause` initialized with `this`, not with `null`. But using `this` as a default argument for `cause` does not work. I'll anyway propose some related fixes in a rework of PR #857.

Reply:
From the docs of `Throwable.cause`:

```
The throwable that caused this throwable to get thrown, or null if this
throwable was not caused by another throwable, or if the causative
throwable is unknown.  If this field is equal to this throwable itself,
it indicates that the cause of this throwable has not yet been
initialized.
```

So setting it to null is fine.

Reply:
The version with `constructor(message: String?, cause: Exception? = null)` still did not work, see https://travis-ci.com/heremaps/oss-review-toolkit/builds/84479452#L1075, the reason being that we do explicitly initialize `cause` with `null` in this case, and then the call to `initCause` at https://github.com/heremaps/oss-review-toolkit/blob/45e623b547f1d34cbf101678a79c5091b4ff7ec8/downloader/src/main/kotlin/Downloader.kt#L151 failed because we cannot initialize `cause` again. So we should explicitly leave it uninitialized if no cause is provided instead of setting it to `null`.